### ISSUE TITLE
Go paths not saving on Komodo Edit 9 OS X

### ISSUE THREAD

**Issue by leonelquinteros** [Komodo/komodo-go#18]
Open Preferences -> Languages -> Go
"Default Go location" fields are empty. 
Fill them with the correct paths. 
"Apply" and "OK"

Open Preferences -> Languages -> Go again. 
"Default Go location" fields are empty again.

pystderr.log file reports the following: 

```
ERROR:xpcom:Creation of class '<class 'component:koGoLanguage.KoGolangLinter'>' failed!
Exception details follow

ERROR:xpcom:Unhandled exception calling 'int8 * createInstance(in nsISomething, in nsIID &, out retval InterfaceIs *);'
Traceback (most recent call last):
  File "/Applications/Komodo Edit 9.app/Contents/Resources/python/xpcom/server/policy.py", line 300, in _CallMethod_
    return 0, func(*params)
  File "/Applications/Komodo Edit 9.app/Contents/Resources/python/xpcom/server/factory.py", line 57, in createInstance
    return self.klass()
  File "/Users/leo/Library/Application Support/KomodoEdit/9.0/XRE/extensions/golang@activestate.com/components/koGoLanguage.py", line 174, in __init__
    getService(components.interfaces.koIAppInfoEx)
  File "/Applications/Komodo Edit 9.app/Contents/Resources/python/xpcom/components.py", line 195, in getService
    return serviceManager.getServiceByContractID(self.contractid, _get_good_iid(iid))
  File "<XPCOMObject method 'getServiceByContractID'>", line 3, in getServiceByContractID
Exception: 2147746132 (NS_ERROR_FACTORY_NOT_REGISTERED)
[2015-03-27 11:40:11,658] [ERROR] lint: 
-- EXCEPTION START --
[Exception... "Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [koILintService.addRequest]"  nsresult: "0x8000ffff (NS_ERROR_UNEXPECTED)"  location: "JS frame :: chrome://komodo/content/lint/lint.js :: this.lintBuffer.prototype._issueRequest :: line 477"  data: no]
+ toString (function) 3 lines
+ message (string) 'Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [koILintService.addRequest]'
+ result (number) 2147549183
+ name (string) 'NS_ERROR_UNEXPECTED'
+ filename (string) 'chrome://komodo/content/lint/lint.js'
+ lineNumber (number) 477
+ columnNumber (number) 0
+ inner (object) null
| + data (object) null
| + stack (string) 'this.lintBuffer.prototype._issueRequest@chrome://komodo/content/lint/lint.js:477:12
this.lintBuffer.prototype._continueRequest@chrome://komodo/content/lint/lint.js:421:8
'
+ location (object) JS frame :: chrome://komodo/content/lint/lint.js :: this.lintBuffer.prototype._issueRequest :: line 477
| + QueryInterface (function) 3 lines
| + language (number) 2
| + languageName (string) 'JavaScript'
| + filename (string) 'chrome://komodo/content/lint/lint.js'
| + name (string) 'this.lintBuffer.prototype._issueRequest'
| + lineNumber (number) 477
| + columnNumber (number) 12
| + sourceLine (string) ''
| + caller (object) JS frame :: chrome://komodo/content/lint/lint.js :: this.lintBuffer.prototype._continueRequest :: line 421
| + formattedStack (string) 'this.lintBuffer.prototype._issueRequest@chrome://komodo/content/lint/lint.js:477:12
this.lintBuffer.prototype._continueRequest@chrome://komodo/content/lint/lint.js:421:8
'
| + stack
    this.lintBuffer.prototype._issueRequest@chrome://komodo/content/lint/lint.js:477:12
    this.lintBuffer.prototype._continueRequest@chrome://komodo/content/lint/lint.js:421:8
-- EXCEPTION END --
```

**Comment by leonelquinteros** [comment]
After rebuilding the XPI using the new version, error persists, but error log changes to: 

```
[2015-03-27 12:06:17,923] [ERROR] console-logger: NS_ERROR_XPC_CI_RETURNED_FAILURE: Component returned failure code: 0x80570015 (NS_ERROR_XPC_CI_RETURNED_FAILURE) [nsIJSCID.createInstance] (2) in chrome://golang/content/pref/pref-golang.js:98
Traceback (most recent call last):
  File "chrome://golang/content/pref/pref-golang.js", line 98, in 
```

**Comment by cgchoffman** [comment]
Sounds like issue [58](https://github.com/Komodo/KomodoEdit/issues/58) in the KomodoEdit repo.

**Comment by leonelquinteros** [comment]
Thanks for the reference. 

Yes, i'm on OS X Yosemite indeed.

**Comment by toddw-as** [comment]
Resolved with recent commits - will be fixed in 9.0.1:
https://github.com/Komodo/KomodoEdit/issues/58
